pmd-visualforce/src/main/java/net/sourceforge/pmd/lang/vf/rule/security/VfHtmlStyleTagXssRule.java

```diff
@@ -0,0 +1,165 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.vf.rule.security;
+
+import java.util.EnumSet;
+import java.util.regex.Pattern;
+
+import net.sourceforge.pmd.lang.vf.ast.ASTContent;
+import net.sourceforge.pmd.lang.vf.ast.ASTElExpression;
+import net.sourceforge.pmd.lang.vf.ast.ASTElement;
+import net.sourceforge.pmd.lang.vf.ast.ASTText;
+import net.sourceforge.pmd.lang.vf.ast.VfNode;
+import net.sourceforge.pmd.lang.vf.rule.AbstractVfRule;
+import net.sourceforge.pmd.lang.vf.rule.security.lib.ElEscapeDetector;
+
+
+public class VfHtmlStyleTagXssRule extends AbstractVfRule {
```

Comment:
Creating a new rule named VfHtmlStyleTagXss. I've been torn between modifying existing VfUnescapeElRule and adding a new rule. My reasoning to create a new rule:
1. VfUnescapeEl rule has grown over the years and looks pretty crowded now. Adding more conditions would only increase the crowding and may also accidentally cause regressions.
2. Unique rule violation messages for individual cases would be more informative to users.
3. A new separate rule would be easier to turn off if a user wanted to get violations only from VfUnescapeEl rule.

Please let me know your thoughts on this. Thanks!